.github/workflows/haskell.yml

```diff
@@ -46,6 +46,19 @@ jobs:
         os: [ubuntu-latest]
 
     steps:
+
+    - run: |
+        sudo apt-get update
+    - name: Free disk space
```

Comment:
Can we please add a comment here that explains the rational for this change?

If I'm just reading it as it is, I'm really confused as to why we do this here and how we end up running out of diskspace.

I'm not questioning that we are. But we should capture the analysis as to why here as well.

Reply:
Sure, I'll add a comment.  
Basically we can't build nix derivations (which we only do in nightly builds) because `error: writing to file: No space left on device` 

For example here: https://github.com/teodanciu/cardano-ledger/actions/runs/6033766318/job/16370965979#step:4:5647

Admittedly there might be better ways to deal with this, going at the root of the problem (maybe cleaning unnecessary things that we create ourselves, though these don't seem to be that big)